crates/trycp_server/src/startup.rs

```diff
@@ -83,6 +94,46 @@ pub fn startup(id: String, log_level: Option<String>) -> Result<(), Error> {
             .read_exact(&mut [0])
             .context(CheckLairReady)?;
 
+            match lair_shim {
+                Some(delay) => {
+                    const SHIM_FILE: &str = "shim/socket";
+                    const LAIR_FILE: &str = "keystore/socket";
+                    std::fs::create_dir_all(player_dir.join("shim")).with_context(|| CreateDir {
+                        path: player_dir.join("shim").clone(),
+                    })?;
+                    let lair_stdout_log_path = player_dir.join(SHIM_STDERR_LOG_FILENAME);
+                    let mut shim = Command::new("lair-shim")
+                        .current_dir(&player_dir)
+                        .arg("-p")
+                        .arg(player_dir.join(SHIM_FILE))
+                        .arg("-l")
+                        .arg(player_dir.join(LAIR_FILE))
+                        .arg("-t")
+                        .arg(delay.to_string())
+                        .stdout(Stdio::piped())
+                        .stderr(
+                            std::fs::OpenOptions::new()
+                            .append(true)
+                            .create(true)
+                            .open(&lair_stdout_log_path)
+                            .context(CreateLairShimStdoutFile {
+                                path: lair_stdout_log_path,
+                            })?,
+                        )
+                        .spawn()
+                        .context(SpawnShim)?;
+
+                    // Wait until shim begins to output before starting conductor,
+                    // otherwise Holochain starts its own copy of shim that we can't manage.
+                    shim.stdout
+                        .as_mut()
+                        .unwrap()
+                        .read_exact(&mut [0])
+                        .context(CheckLairReady)?;
+                },
+                None => {}
+            }
```

Comment:
Let's remove some indentations (include previous suggestion)

```suggestion
            if let Some(delay) = lair_shim {
                const SHIM_FILE: &str = "shim/socket";
                const LAIR_FILE: &str = "keystore/socket";
                
                let shim_dir = player_dir.join("shim");
                
                std::fs::create_dir_all(&amp;shim_dir).with_context(|| CreateDir {
                    path: shim_dir,
                })?;
                let lair_stdout_log_path = player_dir.join(SHIM_STDERR_LOG_FILENAME);
                let mut shim = Command::new("lair-shim")
                    .current_dir(&player_dir)
                    .arg("-p")
                    .arg(player_dir.join(SHIM_FILE))
                    .arg("-l")
                    .arg(player_dir.join(LAIR_FILE))
                    .arg("-t")
                    .arg(delay.to_string())
                    .stdout(Stdio::piped())
                    .stderr(
                        std::fs::OpenOptions::new()
                        .append(true)
                        .create(true)
                        .open(&lair_stdout_log_path)
                        .context(CreateLairShimStdoutFile {
                            path: lair_stdout_log_path,
                        })?,
                    )
                    .spawn()
                    .context(SpawnShim)?;
 
                // Wait until shim begins to output before starting conductor,
                // otherwise Holochain starts its own copy of shim that we can't manage.
                shim.stdout
                    .as_mut()
                    .unwrap()
                    .read_exact(&mut [0])
                    .context(CheckLairReady)?;
            }
```